### ISSUE TITLE
Namespaced flow cannot find clusteroutput

### ISSUE THREAD

**Issue by jeremych1000** [kube-logging/logging-operator#1328]
**Describe the bug**:
- I have a Flow in the namespace `foobar-logging`
- I have a ClusterOutput called `global-opensearch` in the namespace `logging`
- In the flow, I reference the output using
```
spec:
  globalOutputRefs:
    - global-opensearch
```
- The opensearch clusteroutput shows as active
- However, the Flow does not turn active, and says
```
Status:
  Active:  false
  Problems:
    dangling global output reference: global-opensearch
  Problems Count:  1
Events:            <none>
```

**Expected behaviour**:
A flow should be able to reference a clusteroutput in a different namespace

**Steps to reproduce the bug**:
- create a flow in one namespace
- create a clusteroutput in another
- flow cannot connect to the clusteroutput

**Additional context**:
controlNamespace = `logging`

**Environment details**:
- Kubernetes version (e.g. v1.15.2): `1.26`
- Cloud-provider/provisioner (e.g. AKS, GKE, EKS, PKE etc): `EKS`
- logging-operator version (e.g. 2.1.1): `4.1.0`
- Install method (e.g. helm or static manifests): helm chart
- Logs from the misbehaving component (and any other relevant logs): x
- Resource definition (possibly in YAML format) that caused the issue, without sensitive data: 

clusteroutput
```
apiVersion: logging.banzaicloud.io/v1beta1
kind: ClusterOutput
metadata:
  name: global-opensearch
  namespace: logging
spec:
  opensearch:
    <redacted>
```

if I describe the clusteroutput,
```
status:
  active: true
```

flow
```
apiVersion: logging.banzaicloud.io/v1beta1
kind: Flow
metadata:
  name: foobar-logging-flow
  namespace: foobar-logging
spec:
  globalOutputRefs:
    - global-opensearch
  filters:
    - stdout:
        output_type: json
    - record_modifier:
        records:
          - "a.cluster": "redacted"
          - "a.region": "us-east-1"
          - "a.env": "dev"
    - tag_normaliser:
        format: ${namespace_name}.${pod_name}.${container_name}
  match:
    - exclude:
        labels:
          app: fluentbit
    - select: {}
```

/kind bug

**Comment by pepov** [comment]
How does your logging resource look like exactly? Do you perhaps have the watchNamespaces set to something specific?

**Comment by pepov** [comment]
No, it should work both ways. What does your logging resource look like?

**Comment by jeremych1000** [comment]
> No, it should work both ways. What does your logging resource look like?

See https://github.com/kube-logging/logging-operator/issues/1328#issuecomment-1543696019

Is splitting out fluentbit into another logging resource a supported use case? I run a multi-tenant cluster and want to be able to turn logging on and off per tenant i.e. only install fluentbit and the Flow if the tenant wants logging.

**Comment by pepov** [comment]
Sorry, I was from mobile and didn't see your previous comments. 

> Is splitting out fluentbit into another logging resource a supported use case? I run a multi-tenant cluster and want to be able to turn logging on and off per tenant i.e. only install fluentbit and the Flow if the tenant wants logging.

No that setup is not supported at the moment, but I'm going to work on this exact scenario soon. 

> If I have the Flow pointing to a ClusterFlow in the same namespace it works. If the ClusterFlow is in another namespace it doesn't work.

It does work for me with logging operator chart `4.1.0`:
```
kubectl create ns logging
kubectl create ns tenant

kubectl apply -f - <<"EOF"
apiVersion: logging.banzaicloud.io/v1beta1
kind: Logging
metadata:
  name: default
spec:
  fluentd: {}
  fluentbit: {}
  controlNamespace: logging
---
apiVersion: logging.banzaicloud.io/v1beta1
kind: ClusterOutput
metadata:
  name: cluster-out
  namespace: logging
spec:
  nullout: {}
---
apiVersion: logging.banzaicloud.io/v1beta1
kind: Flow
metadata:
  name: flow
  namespace: tenant
spec:
  selectors:
    sampleKey: sampleValue
  globalOutputRefs:
    - "cluster-out"
EOF
```

```
k get logging-all -A
NAMESPACE   NAME                                               ACTIVE   PROBLEMS
logging     clusteroutput.logging.banzaicloud.io/cluster-out   true

NAMESPACE   NAME                                     AGE
            logging.logging.banzaicloud.io/default   3m41s

NAMESPACE   NAME                               ACTIVE   PROBLEMS
tenant      flow.logging.banzaicloud.io/flow   true
```

**Comment by pepov** [comment]
```
apiVersion: logging.banzaicloud.io/v1beta1
kind: Flow
metadata:
  name: foobar-logging-flow
  namespace: foobar-logging
spec:
  globalOutputRefs:
    - global-opensearch
  filters:
    - stdout:
        output_type: json
    - record_modifier:
        records:
          - "a.cluster": "redacted"
          - "a.region": "us-east-1"
          - "a.env": "dev"
    - tag_normaliser:
        format: ${namespace_name}.${pod_name}.${container_name}
  match:
    - exclude:
        labels:
          app: fluentbit
    - select: {}
```

you should add the `tag_normaliser` as the first item in the `filters` sequence as it reemits the event and the other two filters will run twice

**Comment by pepov** [comment]
Closing this because the lack of activity, please reopen if needed

**Comment by konstantin-schura** [comment]
The same issue with SyslogNGFlow and SyslogNGClusterOutput.
version: 4.2.3

apiVersion: logging.banzaicloud.io/v1beta1
kind: SyslogNGClusterOutput
metadata:
  name: elk
  namespace: logging
spec:
  syslog:
    host: logstash
    port: 5000
    template: "$(format-json
                  --subkeys json.
                  --exclude json.kubernetes.labels.*
                  json.kubernetes.labels=literal($(format-flat-json --subkeys json.kubernetes.labels.)))\n"
    transport: tcp

apiVersion: logging.banzaicloud.io/v1beta1
kind: SyslogNGFlow
metadata:
  name: syslog-flow
  namespace: nginx
spec:
  match:
    or:
      - regexp:
          value: json.kubernetes.labels.app
          pattern: nginx
          type: string
  globalOutputRefs:
    - elk

As result I have this config in syslog-ng

destination "clusteroutput_logging_elk" {
    syslog("logstash" port(5001) transport("tcp") template("$(format-json --subkeys json. --exclude json.kubernetes.labels.* json.kubernetes.labels=literal($(format-flat-json --subkeys json.kubernetes.labels.)))\n") persist_name("clusteroutput_logging_elk"));
};

log {
    source("main_input");
    destination("clusteroutput_nginx_syslog-flow");
};


There is destination  **clusteroutput_logging_elk**, but in section **log** it points to  wrong clusteroutput **clusteroutput_nginx_syslog-flow**.

**Comment by pepov** [comment]
@konstantin-schura thanks for reporting! this issue is valid, but has already been fixed in version 4.3.0
